src/components/analyses/toolbar/AnalysesDotMenu.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * @author sriram
+ *
+ * A dot menu for Analyses View.
+ *
+ */
+
+import React from "react";
+
+import ids from "../ids";
+import messages from "../messages";
+import {
+    isInteractive,
+    allowAnalysisTimeExtn,
+    isBatchAnalysis,
+} from "../utils";
+
+import { build, DotMenu, getMessage, withI18N } from "@cyverse-de/ui-lib";
+import { ListItemIcon, ListItemText, MenuItem } from "@material-ui/core";
+
+import {
+    Info,
+    HourglassEmptyRounded as HourGlassIcon,
+    Launch as LaunchIcon,
+    PermMedia as OutputFolderIcon,
+    Repeat as RelaunchIcon,
+    UnfoldMore as UnfoldMoreIcon,
+} from "@material-ui/icons";
+
+function AnalysesDotMenu(props) {
+    const {
+        baseId,
+        ButtonProps,
+        username,
+        onDetailsSelected,
+        detailsEnabled,
+        selected,
+        getSelectedAnalyses,
+        handleInteractiveUrlClick,
+        handleGoToOutputFolder,
+        handleBatchIconClick,
+    } = props;
+
+    const isSingleSelection = selected?.length === 1;
+    const selectedAnalyses = getSelectedAnalyses ? getSelectedAnalyses() : null;
+
+    let isBatch = false,
+        isVICE = false,
+        allowTimeExtn = false,
+        allowRelaunch = false;
+
+    if (isSingleSelection && selectedAnalyses) {
+        isBatch = isBatchAnalysis(selectedAnalyses[0]);
+        isVICE = isInteractive(selectedAnalyses[0]);
+        allowTimeExtn = allowAnalysisTimeExtn(selectedAnalyses[0], username);
+        allowRelaunch = !selectedAnalyses[0]?.app_disabled;
+    }
+
+    return (
+        <DotMenu
+            baseId={baseId}
+            ButtonProps={ButtonProps}
+            render={(onClose) => [
+                detailsEnabled && (
+                    <MenuItem
```

Comment:
One thing that could be done to shrink the size of this function is to split out the MenuItems into separate, unexported React components in the same file. Personally I try to keep my functions/components down to about a single screen or slightly larger.